pkg/ansible/flags/flag.go

```diff
@@ -59,7 +61,16 @@ func (f *Flags) AddTo(flagSet *pflag.FlagSet, helpTextPrefix ...string) {
 	flagSet.IntVar(&f.MaxWorkers,
 		"max-workers",
 		runtime.NumCPU(),
-		"Maximum number of workers to use. Overridden by environment variable.",
+		strings.Join(append(helpTextPrefix,
+			"(Deprecated) Maximum number of workers to use. Use max-concurrent-reconciles instead."),
```

Comment:
Rather than saying `(Deprecated)` in the usage text, we should be able to define the flag as normal, but then use either `flagSet.MarkDeprecated("max-workers", ...)` or `flagSet.MarkHidden("max-workers")` depending on whether we want to handle the deprecation message ourselves.

In this case, it probably does make sense to do the deprecation message ourselves because we can tailor it depending on whether `--max-concurrent-reconciles` is set.